pkg/cluster/cluster.go

```diff
@@ -37,33 +38,32 @@ const (
 	PeerTypeQuery = "query"
 )
 
-// NewPeer creates a new peer.
-func NewPeer(
-	l log.Logger,
-	name string,
-	typ PeerType,
-	addr string,
-	advertiseAddr string,
-	apiAddr string,
-	knownPeers []string,
-) (*Peer, error) {
+type JoinConfig struct {
+	Name          string
+	Addr          string
+	AdvertiseAddr string
+	KnownPeers    []string
+}
+
+// Join creates the peer and joins it to the cluster with the given state.
+func Join(l log.Logger, joinCfg JoinConfig, state PeerState, reg *prometheus.Registry) (*Peer, error) {
```

Comment:
Move the registry after logger maybe? To me those and contexts always belong in the front.